packages/core-data/src/selectors.js

```diff
@@ -153,6 +153,20 @@ export function getEntityRecords( state, kind, name, query ) {
 	return getQueriedItems( queriedState, query );
 }
 
+/**
+ * Returns the cached Entity's records (doesn't trigger the resolver).
+ *
+ * @param {Object}  state  State tree
+ * @param {string}  kind   Entity kind.
+ * @param {string}  name   Entity name.
+ * @param {?Object} query  Optional terms query.
+ *
+ * @return {Array} Records.
+ */
+export function getCachedEntityRecords( state, kind, name, query ) {
+	return getEntityRecords( state, kind, name, query );
```

Comment:
Hm, this relies on a behavior which historically I've considered an unintentional oversight, that calling these selector functions directly won't trigger a resolver because the resolver has to be triggered through the `select` function. We've worked around in the past by defining resolvers for anything which derives from another selector ([example](https://github.com/WordPress/gutenberg/blob/2fd13b663ef82778e0f4fcfe963ff178fa0e0b6d/packages/core-data/src/resolvers.js#L198-L209)), but this is very tedious and potentially error-prone. Ideally all of the dependencies of a selector could resolve automatically.

I'm not sure what alternative we have, since again, these things are just functions and not inherently hookable (except by forcing a developer to go through `createRegistrySelector`). But I'm flagging it since your changes here tie us further to the status quo.

Reply:
Have you thought about allowing selectors to optionally declare resolver dependencies:

```js
getAutosave.resolvers = [ getAutosaves ];
```

Reply:
> Have you thought about allowing selectors to optionally declare resolver dependencies?:
> 
> ```js
> getAutosave.resolvers = [ getAutosaves ];
> ```

The idea is nice, but in practice it becomes an issue in: How do we pass arguments through to `getAutosaves`? See also https://github.com/WordPress/gutenberg/pull/21467#issuecomment-612154486, which explores another alternative approach.